src/plugins/validation/2and3/semantic-validators/responses.js

```diff
@@ -36,9 +40,9 @@ module.exports.validate = function({ jsSpec, isOAS3 }, config) {
                       i < mediaType.schema[schemaType].length;
                       i++
                     ) {
-                      const hasInlineSchema = !mediaType.schema[schemaType][i]
-                        .$ref;
-                      if (hasInlineSchema) {
+                      const currentSchema = mediaType.schema[schemaType][i];
+                      const hasInlineSchema = !currentSchema.$ref;
+                      if (hasInlineSchema && !arrayItemsAreRef(currentSchema)) {
```

Comment:
While this does address the problem as described in the issue, I don't think it is quite what we want.

In addition to allowing an array of $ref, I think we probably should also allow an array of primitive types.  That is, I do not think we should flag a response schema that is an array of `string` or an array of `integer`.

Reply:
Agreed, though this will likely be more verbose than it would be in the generator (no `isPrimitiveType` to rely on). Perhaps you could write a utility function called isPrimitiveType or something like that so we can re-use it if this comes up again